packages/one-app-bundler/webpack/loaders/common.js

```diff
@@ -39,30 +39,73 @@ const cssLoader = ({ name = '', importLoaders = 2 } = {}) => ({
 
 /* eslint-disable inclusive-language/use-inclusive-words --
 config options for a third party library */
-const purgeCssLoader = () => {
-  const configOptions = getConfigOptions();
-  const whitelistPatterns = configOptions.purgecss.whitelistPatterns
-    ? configOptions.purgecss.whitelistPatterns.map((pattern) => new RegExp(pattern, 'i'))
+// transform strings deep and greedy parameters into regex
+const reconcileSafeList = (safelist) => {
+  const configOptionsReconciled = safelist;
+
+  const greedy = safelist.greedy
+    ? safelist.greedy.map((pattern) => new RegExp(pattern, 'i'))
     : [];
-  const whitelistPatternsChildren = configOptions.purgecss.whitelistPatternsChildren
-    ? configOptions.purgecss.whitelistPatternsChildren.map((pattern) => new RegExp(pattern, 'i'))
+  configOptionsReconciled.greedy = greedy;
+  const deep = safelist.deep
+    ? safelist.deep.map((pattern) => new RegExp(pattern, 'i'))
     : [/:global$/];
-  if (configOptions.purgecss.disabled) return [];
+  configOptionsReconciled.deep = deep;
+
+  return configOptionsReconciled;
+};
+
+const purgeCssLoader = () => {
```

Comment:
an idea to make the code more readable is to have two different functions, `reconcileSafeList` when `safelist` is present and an additional function when it isn't present and `whitelist` needs to be used.

Reply:
check again 😄